pkg/reconciler/v1alpha1/revision/resources/queue.go

```diff
@@ -71,7 +71,7 @@ var (
 
 // makeQueueContainer creates the container spec for queue sidecar.
 func makeQueueContainer(rev *v1alpha1.Revision, loggingConfig *logging.Config, autoscalerConfig *autoscaler.Config,
-	controllerConfig *config.Controller) *corev1.Container {
+	controllerConfig *config.Controller, userPortEnv *corev1.EnvVar) *corev1.Container {
```

Comment:
Instead of using `userPortEnv *corev1.EnvVar` as input, I would use `extraEnvs []corev1.EnvVar`, and then directly append `extraEnvs` into the `Env` list of the queue container.